### DISCUSSION TITLE
Have the ability to change/create a different app layout/folder architecture

### DISCUSSION THREAD

**Discussion by AlexandrosMelissas** [loopbackio/loopback-next#7960] [Ideas]
Hello. I've just tried the framework and it seems awesome. There is only 1 drawback for me. Due to big scalability issues, would it be possible if we could alternate to this layout or similar to it?
![image](https://user-images.githubusercontent.com/56912034/135648268-55c829e2-87b2-4648-ab22-0f5e57fab3b4.png)

Instead of having general folders of models, controllers, services etc , each entity-concept-module (User | Post | Comment etc) would be a folder and inside of it, there would be controllers, services, models etc of that module. 

My suggestion is to either add an option on lib4 app creation or be able to change it via cli command.

**Comment by dhmlau** [comment]
IIRC there were comments about having subfolders within the model/controller/services folder like below (unfortunately I couldn't find the comment/suggestion for now). Thoughts? 
i.e. 
/models
    /category1-models
         a1.model.ts
         a2.model.ts
     /category2-models
         b1.model.ts
         b2.model.ts
     /...
/controllers
   /category1-controllers
   /category2-controllers

**Reply by AlexandrosMelissas** [reply]
Seems good and it does seem similar to the current one that's used but in my case, and i guess in many other people's cases, it doesn't solve the fact that if we have, let's say, 25 modules, you get lost very easily. However, if you have the module itself as a folder, it's very easy to navigate, most of the times you need multiple things from a specific module (controller + service + model or controller + dto for example) for fixes, debugging or new features.

**Reply by mrmodise** [reply]
@AlexandrosMelissas have a look at https://loopback.io/doc/en/lb4/Creating-components.html you might find it useful for larger applications
